File: firstAlgorithm.py
import random
import math

#feel free to change the number of test cases
TEST_CASE = 10 ** 6

#consider unit circle,
#we create random points with x in (-1,1) interval , y in (-1,1) interval
#if the point is in the unit circle, we increment successor by 1
successor = 0
for i in range(TEST_CASE):
    px = random.uniform(-1.0,1.0)
    py = random.uniform(-1.0,1.0)

    if( px*px + py*py <= 1.0 ):
        successor += 1


#area of unit circle is pi, area of possible point space is 4
#we expect to see that 
# pi/4 = number_of_points_in_circle / number_of_all_points

estimated_pi =  4 * successor / TEST_CASE
print("Estimation:",estimated_pi)

print(f"Error is { round( abs(math.pi - estimated_pi) / math.pi * 100,6 ) } %")

File: firstAlgorithm.cpp
#include <iostream>
#include <stdlib.h>
#include <ctime>
#include <iomanip>
#include <cmath>


using std::cout;

// returns a number between -1 and 1
double random_number()
{
    //rand / randmax is between 0 and 1. Mulptiply by 2 then the interval is (0,2). Subtract 1 and the interval is -1,1.
    return ( (double) rand() / RAND_MAX) * 2 - 1;
}

const long TEST_CASE = 1e8;

int main()
{
    srand(time(0));

    
    long inside = 0; //number of points inside the unit circle.
    for(long i = 0; i < TEST_CASE; i++)
    {
        double px = random_number();
        double py = random_number();

        if(px*px + py*py <= 1) // unit circle is x^2 + y^2 <= 1
            inside++;
    }

    double estimation = 4.0 * inside / TEST_CASE;
    cout << "Estimation: " << estimation << "\n";

    cout << std::fixed;
    cout << std::setprecision(6);
    cout << "Error is " << abs(M_PI - estimation)/M_PI * 100 <<  " %\n";

    return 0;
}